allennlp/commands/__init__.py

```diff
@@ -51,37 +50,33 @@ def add_argument(self, *args, **kwargs):
 
 
 def create_parser(
-    prog: str = None, subcommand_overrides: Dict[str, Subcommand] = None
+    prog: str = None, subcommand_overrides: Iterable[Subcommand] = None
 ) -> argparse.ArgumentParser:
     """
     Creates the argument parser for the main program.
     """
-    if subcommand_overrides is None:
-        subcommand_overrides = {}
+    subcommand_overrides = subcommand_overrides or []
 
-    parser = ArgumentParserWithDefaults(description="Run AllenNLP", usage="%(prog)s", prog=prog)
-    parser.add_argument("--version", action="version", version="%(prog)s " + __version__)
+    parser = ArgumentParserWithDefaults(description="Run AllenNLP", prog=prog)
+    parser.add_argument("--version", action="version", version=f"%(prog)s {__version__}")
 
     subparsers = parser.add_subparsers(title="Commands", metavar="")
 
-    subcommands = {
-        # Default commands
-        "train": Train(),
-        "evaluate": Evaluate(),
-        "predict": Predict(),
-        "elmo": Elmo(),
-        "fine-tune": FineTune(),
-        "dry-run": DryRun(),
-        "make-vocab": DryRun(),  # deprecated, but keeping for backward compatibility.
```

Comment:
I removed "make-vocab" implicitly. Please, lmk if you wanted it to be added again.